docs/zkapps/advanced-snarkyjs/time-locked-accounts.mdx

```diff
@@ -0,0 +1,114 @@
+---
+title: Time-Locked Accounts
+hide_title: true
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Time-Locked Accounts
+
+Time-locking allows you to pay someone MINA or other tokens, which are subject to a vesting schedule. This means that the tokens are initially locked, and only become available for withdrawal after a certain time, or gradually according to a certain schedule.
+
+The zkApp feature that enables time-locking is the `timing` field, which is present on every account. It look like this:
+
+```ts
+type Account = {
+  // ...
+  timing: {
+    isTimed: Bool;
+    initialMinimumBalance: UInt64;
+    cliffTime: UInt32;
+    cliffAmount: UInt64;
+    vestingPeriod: UInt32;
+    vestingIncrement: UInt64;
+  };
+};
+```
+
+`isTimed` tells us whether this account is subject to any time-locking. The other fields are parameters that allow us to define a vesting schedule in a very flexible manner. We will explain them in detail below. By default, an account has `isTimed: false`; in that case, the other fields are dummy values.
```

Comment:
> `isTimed` indicates whether this account is time locked. .... By default, accounts are not time locked, and `isTimed` will be `false` and all other properties will contain nullish values.

Can we improve the last two words even more? Are the others all set to falsy values of `0`?

Reply:
No I think the defaults will be
```ts
{
  initialMinimumBalance: 0,
  cliffTime: 0,
  cliffAmount: 0,
  vestingPeriod: 1,
  vestingIncrement: 0,
}
```

See: https://github.com/MinaProtocol/mina/blob/be9a92606cf395aadd5b67b4cb309d07fad54bb4/src/lib/mina_base/account_timing.ml#L93

I guess developers don't need to care about the values when no timing is set; we could use "default values" if you think "dummy" isn't good style?

Reply:
> we could use "default values"
+1 Cool, that sounds better.